torch/library.h

```diff
@@ -286,59 +368,131 @@ class CAFFE2_API Library final {
+    // catch all def
     CppFunction f(std::forward<Func>(raw_f));
     return _impl(name, std::move(f));
   }
 
-  // Convenience overload for directly specifying the dispatch key.  Dispatch
-  // can validly be either DeviceType or DispatchKey; check torch::dispatch for
-  // the canonical list of accepted overloads.
+  /// \private
+  ///
+  /// Convenience overload for directly specifying the dispatch key when
+  /// impl().  You probably don't need this; instead, prefer specifying
+  /// the dispatch key for the entire block in TORCH_LIBRARY_IMPL()
   template <typename Dispatch, typename Func>
   Library& impl(const char* name, Dispatch&& key, Func&& raw_f) & {
     return impl(name, dispatch(std::forward<Dispatch>(key), std::forward<Func>(raw_f)));
   }
 
-  // Convenience overload for unboxed only kernels.  These are quite common
-  // but will be eventually eliminated; this function makes it easy to grep for
-  // them.
-  //
-  // TODO: Remove this overload once the makeUnboxedOnly incidence rate
-  // goes way down
+  /// \private
+  ///
+  /// Convenience overload for unboxed only kernels; kernels whose type
+  /// signatures are not supported by our template based metaprogramming
+  /// system.  These are currently quite common but will be eventually
+  /// eliminated.
+  ///
+  /// This is equivalent to calling CppFunction::makeUnboxedOnly() on
+  /// the function, but this name for the function makes it easy to grep for.
   template <typename Func>
   Library& impl_UNBOXED(const char* name, Func* raw_f) & {
+    // TODO: Remove this overload once the makeUnboxedOnly incidence rate
+    // goes way down
     return impl(name, CppFunction::makeUnboxedOnly(raw_f));
   }
 
-  // Register a fallback implementation for all operators which will be used
-  // if there is not a specific implementation for an operator available.
-  // Providing a DispatchKey is MANDATORY for fallback at the moment; e.g.,
-  // only call this from TORCH_LIBRARY_IMPL
+  /// Register a fallback implementation for all operators which will be used
+  /// if there is not a specific implementation for an operator available.
+  /// There MUST be a DispatchKey associated with a fallback; e.g.,
+  /// only call this from TORCH_LIBRARY_IMPL() with namespace `_`.
+  ///
+  /// \param raw_f The function that implements the fallback.  Unboxed
+  ///   functions typically do not work as fallback functions, as
+  ///   fallback functions must work for every operator (even though
+  ///   they have varying type signatures).  Typical arguments are
+  ///   CppFunction::makeFallthrough() or
+  ///   CppFunction::makeFromBoxedFunction()
+  ///
+  /// ```
+  /// // Example:
+  ///
+  /// void xla_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
+  ///   // ... see aten/src/ATen/core/dispatch/backend_fallback_test.cpp
+  ///   // for a full example ...
+  /// }
+  ///
+  /// TORCH_LIBRARY_IMPL(_, XLA, m) {
```

Comment:
How about we use the pytorch/xla code snippet directly here, it seems more straightforward (I always forget what `fallback` and `makeFallthrough` are used for but this snippet explains it pretty well ;) 
```
  TORCH_LIBRARY_IMPL(_, XLAPreAutograd, m) {
    m.fallback(torch::CppFunction::makeFallthrough());
  }
```